### PR TITLE
Edit functionality

### PR DESCRIPTION
**Pull request by rickifox**
Added book information edition functionality

### CODE REVIEW

**Comment by MorganeGrenes** [comment]
Ca m'a l'air pas mal 👍 😸

**Review by MorganeGrenes — DISMISSED** [review]

**Review by YannRaulet — DISMISSED** [review]
It's all good !!

**Review by MorganeGrenes — DISMISSED** [review]

File: config/config.php [hunk 1]
```diff
@@ -17,4 +17,4 @@
 define('APP_VIEW_PATH', __DIR__ . '/../src/View/');
 define('APP_CACHE_PATH', __DIR__ . '/../temp/cache/');
 
-define('HOME_PAGE', 'home/index');
+define('HOME_PAGE', 'Livre/index');
```
Comment (RomainClair) [inline config/config.php]: Livre should probably be livre

File: src/Controller/LivreController.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: root
+ * Date: 11/10/17
+ * Time: 16:07
+ * PHP version 7
+ */
+
```
Comment (RomainClair) [inline src/Controller/LivreController.php]: Really ?

File: src/Model/LivreManager.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: sylvain
+ * Date: 07/03/18
+ * Time: 18:20
+ * PHP version 7
+ */
```
Comment (RomainClair) [inline src/Model/LivreManager.php]: Edit the comments too when you copy/paste things

File: src/Model/LivreManager.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+
+    /**
+     * @param array $livre
+     * @return int
+     */
+    public function insert(array $livre): int
+    {
+        // prepared request
+        $statement = $this->pdo->prepare("INSERT INTO livre (titre, auteur, parution,
```
Comment (RomainClair) [inline src/Model/LivreManager.php]: You do have a const with the table name : use it

File: src/Model/LivreManager.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        :lecture, :lu, :isbn, :localisation, :genre, :description)");
+        $statement->bindValue('titre', $livre['titre'], PDO::PARAM_STR);
+        $statement->bindValue('auteur', $livre['auteur'], PDO::PARAM_STR);
+        $statement->bindValue('parution', $livre['parution'], PDO::PARAM_INT);
+        $statement->bindValue('lecture', $livre['lecture'], PDO::PARAM_INT);
+        $statement->bindValue('lu', $livre['lu'], PDO::PARAM_BOOL);
+        $statement->bindValue('isbn', $livre['isbn'], PDO::PARAM_STR);
+        $statement->bindValue('localisation', $livre['localisation'], PDO::PARAM_STR);
+        $statement->bindValue('genre', $livre['genre'], PDO::PARAM_STR);
+        $statement->bindValue('description', $livre['description'], PDO::PARAM_STR);
```
Comment (RomainClair) [inline src/Model/LivreManager.php]: This method calls could fail and return false. You should check it to avoid having PHP errors : method calls on false.

**Review by RomainClair — CHANGES_REQUESTED** [review]
You should not modifiy your framework.
The AbstractManager is part of the framework, you probably should keep it unmodified

The git workflow defined for this project stands that you'll have to do one branch per user story/functionnality.

This branch is a lots of functionnality branch. Long to review, long to debug. You should avoid this.

**Review by YannRaulet — DISMISSED** [review]
OK for the changes !!

**Review by MorganeGrenes — DISMISSED** [review]

**Review by YannRaulet — DISMISSED** [review]

**Review by AurelieFqt — DISMISSED** [review]

**Review by MorganeGrenes — APPROVED** [review]

**Review by axelhamil — COMMENTED** [review]
Good

**Review by axelhamil — APPROVED** [review]

**Pull request closed without merging**